src/value/table.ts

```diff
@@ -0,0 +1,79 @@
+import * as prismic from "@prismicio/client";
+
+import { createFaker, Faker } from "../lib/createFaker";
+
+import { MockValueStateConfig, MockValueConfig } from "../types";
+import { richText } from "./richText";
+
+export type MockTableValueConfig<
+	Model extends
+		prismic.CustomTypeModelTableField = prismic.CustomTypeModelTableField,
+	State extends prismic.FieldState = prismic.FieldState,
+> = MockValueConfig<Model> & MockValueStateConfig<State>;
+
+export type MockTableValue<
+	State extends prismic.FieldState = prismic.FieldState,
+> = prismic.TableField<State>;
+
+export const table = <
+	Model extends
+		prismic.CustomTypeModelTableField = prismic.CustomTypeModelTableField,
+	State extends prismic.FieldState = "filled",
+>(
+	config: MockTableValueConfig<Model, State>,
+): MockTableValue<State> => {
+	const faker = config.faker || createFaker(config.seed);
+
+	return (
+		config.state === "empty"
+			? null
+			: {
+					head: {
+						rows: [
+							{
+								cells: Array.from({ length: 3 }, () =>
+									mockTableCell({
+										type: "header",
+										faker,
+									}),
+								),
+							},
+						],
+					},
+					body: {
+						rows: Array.from({ length: 3 }, () => ({
+							cells: Array.from({ length: 3 }, () =>
+								mockTableCell({
+									type: "data",
+									faker,
+								}),
+							),
+						})),
+					},
+				}
+	) as MockTableValue<State>;
```

Comment:
This isn't the format we're using anymore and will need to be updated with a newer version of `client` that uses a newer version of `types-internal`, I think.

Reply:
are we sure about this? this is v2 format and not types internal, right?

Reply:
You're right, you can forget this comment.